tmk_core/common/action_layer.c

```diff
@@ -296,15 +296,15 @@ action_t store_or_get_action(bool pressed, keypos_t key) {
  *
  * Gets the layer based on key info
  */
-int8_t layer_switch_get_layer(keypos_t key) {
+uint8_t layer_switch_get_layer(keypos_t key) {
 #ifndef NO_ACTION_LAYER
   action_t action;
   action.code = ACTION_TRANSPARENT;
 
   uint32_t layers = layer_state | default_layer_state;
   /* check top layer first */
   for (int8_t i = 31; i >= 0; i--) {
```

Comment:
It's good you didn't change `i` to `uint8_t`, that's a disaster waiting to happen.

LGTM

Reply:
Well, I did, then I thought about it 😁

Reply:
I'm curious, why is that?

Reply:
Oh, wait, I see it now.

Reply:
The for loop counts from 31 to 0, and stops when *i* is < 0. But unsigned integers can never be negative, so when *i* is zero and decremented again, it wraps around to 255 -- which is still >= 0. Thus we get an endless loop, most likely locking up the keyboard until you press the physical reset button (if you have one) to get into the bootloader. You can't even reset with the keycode because this function is called on every keypress to determine whether it's transparent!

Reply:
Okay, so signed is positive or negative, but unsigned is always positive. Same result I was thinking, but wrong cause.

Thanks @fauxpark.